src/hssm/hssm.py

```diff
@@ -592,35 +614,120 @@ def sample_posterior_predictive(
+                    setattr(idata_copy, "posterior", tmp_posterior)
+                    self.model.predict(
+                        idata_copy, kind, data, True, include_group_specific
+                    )
+                    posterior_predictive_list.append(idata_copy["posterior_predictive"])
+
+                if inplace:
+                    idata.add_groups(
+                        posterior_predictive=xr.concat(
+                            posterior_predictive_list, dim="draw"
+                        )
+                    )
+                    # for inplace, we don't return anything
+                    return None
+                else:
+                    # Reassign original posterior to idata_copy
+                    setattr(idata_copy, "posterior", idata["posterior"])
+                    # Add new posterior predictive group to idata_copy
+                    del idata_copy["posterior_predictive"]
+                    idata_copy.add_groups(
+                        posterior_predictive=xr.concat(
+                            posterior_predictive_list, dim="draw"
+                        )
+                    )
+                    return idata_copy
+            elif inplace:
+                # If not safe-mode
+                # We call .predict() directly without any
+                # chunking of data.
+
+                # .predict() is called on the copy of idata
+                # since we still subsampled (or assigned) the draws
                 self.model.predict(idata_copy, kind, data, True, include_group_specific)
+
+                # posterior predictive group added to idata
                 idata.add_groups(
                     posterior_predictive=idata_copy["posterior_predictive"]
                 )
-
+                # don't return anything if inplace
                 return None
-
+            else:
+                # Not safe mode and not inplace
+                # Function acts as very thin wrapper around
+                # .predict(). It just operates on the
+                # idata_copy object
+                return self.model.predict(
+                    idata_copy, kind, data, inplace, include_group_specific
+                )
+        elif kind == "mean":
+            # If kind == 'mean', we don't need to run the RV directly,
+            # there shouldn't really be any significant memory issues here,
+            # we can simply ignore settings, since the computational overhead
+            # should be very small --> nudges user towards good outputs.
+            _logger.warning(
+                "The kind argument is set to 'mean', but 'draws' argument "
+                + "is not None: The draws argument will be ignored!"
+            )
             return self.model.predict(
-                idata_copy, kind, data, False, include_group_specific
+                idata, kind, data, inplace, include_group_specific
             )
 
```

Comment:
Add an `else` clause here to throw an error whenever other values are specified?